backend/Match/views.py

```diff
@@ -27,10 +27,10 @@ def get(self, request):
         if token.user_left_side.id != request.user.id:
             return Response({'error': 'You are not the host user in the token'}, status=status.HTTP_403_FORBIDDEN)
 
-        match, player_left, player_right = GameSetupManager.create_match_and_its_players(token)
+        match, player_left, player_right = MatchSetupManager.create_match_and_its_players(token)
```

Comment:
wait, you are throwing errors in the create_match_and ... function, but you are not catching them here, so we are probably anyway throwing a 500 if something goes wrnog there.